### ISSUE TITLE
Loot Chest reward is "NULL" after computer/Enigmatica 2: Expert pack crashed

### ISSUE THREAD

**Issue by agriasrex** [Funwayguy/StandardQuestingPack#96]
While playing Enigmatica 2: Expert, pc crashed so I started the pack up again. Quest book was entirely reset, so I bq_admin edit'd the quests back in, and, to my surprised, all of the "redone" loot chests were "NULL" as the reward when opened, and all new quests that I finished _honestly_ (or granted myself through creative mode) were null.

MultiMC shows this as the log, which indicates **a DIVIDE BY ZERO error:**

`[10:44:05] [Client thread/FATAL] [minecraft/Minecraft]: Error executing task
java.util.concurrent.ExecutionException: java.lang.ArithmeticException: / by zero
	at java.util.concurrent.FutureTask.report(Unknown Source) ~[?:1.8.0_221]
	at java.util.concurrent.FutureTask.get(Unknown Source) ~[?:1.8.0_221]
	at net.minecraft.util.Util.func_181617_a(SourceFile:47) [h.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1088) [bib.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:398) [bib.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:123) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_221]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_221]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_221]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: java.lang.ArithmeticException: / by zero
	at bq_standard.client.gui.GuiLootChest.initPanel(GuiLootChest.java:44) ~[GuiLootChest.class:?]
	at betterquesting.api2.client.gui.GuiScreenCanvas.func_73866_w_(GuiScreenCanvas.java:120) ~[GuiScreenCanvas.class:?]
	at net.minecraft.client.gui.GuiScreen.func_146280_a(GuiScreen.java:478) ~[blk.class:?]
	at net.minecraft.client.Minecraft.func_147108_a(Minecraft.java:1018) ~[bib.class:?]
	at bq_standard.network.handlers.NetLootClaim.onClient(NetLootClaim.java:59) ~[NetLootClaim.class:?]
	at betterquesting.network.PacketQuesting$HandleClient.lambda$onMessage$0(PacketQuesting.java:114) ~[PacketQuesting$HandleClient.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_221]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_221]
	at net.minecraft.util.Util.func_181617_a(SourceFile:46) ~[h.class:?]
	... 17 more`

Is there any way to fix it? I have several dozen hours into this world and can't really restart now, but continuing without rewards from the loot chests isn't any fun.

**Comment by Pligzy88** [comment]
i get NULL even with no crashes on a different mod pack and it has ruined the pack for me now.

**Comment by JFMillerUSA** [comment]
Loot Chest reward is "NULL" in Engineer's Life Pack also

**Comment by agriasrex** [comment]
Two months later, and no update from author? Widely used mod isn't being supported anymore?

**Comment by Funwayguy** [comment]
1) The original crash was already fixed
2) NULL happens because I removed the error case that defaulted to dungeon loot. It was never meant to be used by pack devs but apparently people got lazy and assumed that by not bothering to setup any actual loot tables they'd instead rely on this fallback case.
3) I have since changed "NULL" to instead say "No Loot Setup" to be more clear on why this is the way it is.
4) People are still blaming me for changing a thing that was never meant to be used in the first place and "breaking their packs".
5) I contacted the Engineer's Life pack dev offering to help them resolve this on January 19th but not only have they not replied, they updated the expansion and still didn't bother to add their own loot properly. In fact both the original change and the name change were both detailed in the change logs, apparently neither of which they read or adjusted for.
6) I took some downtime from modding to do some art and focus on studies for a while. The mod is relatively stable in its current state (besides people complaining about user error) and I would have stopped my break to put an update out if there was something serious enough to warrant one.
